lms/views/dashboard/api/assignment.py

```diff
@@ -99,13 +103,16 @@ def course_assignments_metrics(self) -> APIAssignments:
                 h_userids=filter_by_h_userids,
             )
         ).all()
-        assignments = self.request.db.scalars(
-            self.assignment_service.get_assignments(
-                course_id=course.id,
-                instructor_h_userid=current_h_userid,
-                h_userids=filter_by_h_userids,
+        assignments_query = self.assignment_service.get_assignments(
+            course_id=course.id,
+            instructor_h_userid=current_h_userid,
+            h_userids=filter_by_h_userids,
+        )
+        if filter_by_assignment_ids:
```

Comment:
Arguably this filter could be move into the service itself but in this case I reckon the service is doing the heavy lifting of complicated joins etc and here we just tweak the query slightly to for the view.

Reply:
Same discussion over: https://github.com/hypothesis/lms/pull/6436#discussion_r1670249819